### ISSUE TITLE
display progress of image pull

### ISSUE THREAD

**Issue by yee379** [jupyterhub/kubespawner#302]
i unfortunately have some rather large jupyterlab instances that i spin up via the kubespawner. They can take upwards of 5-10 minutes to download onto the kubernetes worker nodes.

i would like to suggest that 

1) a progress bar be provided to the user awaiting the image pull as it downloads
2) the timeout for the jupyterlab spawn be adaptive such that a longer than usual pull time doesn't necessarily timeout and error.

**Comment by consideRatio** [comment]
Ooh I got excited about these ideas that i had not considered thanks for sharing them!

Progress bar for the image is complex, sometimes the pods docker is simply awaiting an imagepull being blocked by another one from another pod, there is also no info to gather about progress in the k8s logs i think.

Hmmm, but perhaps clarify more structured information could be possible like:

these are the images to be pulled before successful startup, then "ok" them when they are and have run (init containers), and the pod non-init containers etc, hmmm

the timeout adjustment would require the JH to have an API allowing for this, i think perhaps that if kubespawner got really smart about the startup process this could be nice but it may be too hard to interpret all kinds of delays wisely, hmmm.

A seed is now planted in my mind on how this could be improvable!

**Comment by consideRatio** [comment]
While I want this as well, it is not something that I know we can do as k8s doesn't provide more details on this, and to try collect such information in other ways is out of scope for this project. Due to that, I'll close this feature request as out of scope.
